docs/source/rest_api/state_delta_websockets.rst

```diff
@@ -0,0 +1,149 @@
+*****************************************
+State Delta Subscriptions via Web Sockets
+*****************************************
+
+As transactions are committed to the block chain, an app developer may be
+interested in receiving events related to the changes in state that result.
+These events, `StateDeltaEvents`, include information about the advance of the
+block chain, as well as state changes that can be limited to specific address
+spaces in the global state.
+
+An application can subscribe to receive these events via a web socket, provided
+by the REST API component.  For example, a single-page JavaScript application may
+open a web socket connection and subscribe to a particular transaction family's
+state values, using the incoming events to re-render portions of the display.
+
+.. note::
+
+   All examples here are written in JavaScript, and assumes the Sawtooth REST
+   API is reachable at `localhost`.
+
+Opening a Web Socket
+====================
+
+The application developer must first open a web socket. This is accomplished
+by using standard means.  In the case of in-browser JavaScript:
+
+.. code-block:: javascript
+
+   let ws = new WebSocket('ws:localhost:8080/subscriptions')
+
+If the REST API is running, it should trigger an event on the web socket's
+`onopen` handler.
+
+Subscribing to State Changes
+============================
+
+In order to subscribe to an address space in the global state, first a message
+needs to be sent on the socket with the list of prefixes. It is a best-practice
+to send this message as part of the web socket's `onopen` handler.  In the
```

Comment:
Can this be done later, outside of the onopen handler? Can a client progressively subscribe to and unsubscribe from a set of prefixes it is managing?